src/stream/src/executor/sort_buffer.rs

```diff
@@ -129,46 +146,52 @@ impl<S: StateStore> SortBuffer<S> {
     }
 
     /// Output a stream that is ordered by the sort key.
-    pub fn iter_until<'a, 'b: 'a>(
-        &'a mut self,
-        watermark_val: &'b ScalarImpl,
-    ) -> impl Iterator<Item = DataChunk> + 'a {
-        let last_watermark = self.last_watermark.replace(watermark_val.clone());
-        let g = move || {
-            let mut data_chunk_builder =
-                DataChunkBuilder::new(self.schema.data_types(), self.chunk_size);
-            // Only records with timestamp greater than the last watermark will be output, so
-            // records will only be emitted exactly once unless recovery.
-            let start_bound = if let Some(last_watermark) = last_watermark.clone() {
-                Bound::Excluded((
-                    // TODO: unsupported type or watermark overflow. Do we have better ways instead
-                    // of unwrap?
-                    last_watermark.successor().unwrap(),
-                    OwnedRow::empty().into(),
-                ))
-            } else {
-                Bound::Unbounded
-            };
-            let end_bound = Bound::Excluded((
-                (watermark_val.successor().unwrap()),
-                OwnedRow::empty().into(),
-            ));
-
-            for (_, (row, _)) in self.buffer.range((start_bound, end_bound)) {
-                if let Some(data_chunk) = data_chunk_builder.append_one_row(row) {
-                    // When the chunk size reaches its maximum, we construct a data chunk and
-                    // send it to downstream.
-                    yield data_chunk;
+    ///
+    /// The output stream will not output the same record except during failover and recovery.
+    /// However, if you want to delete one record permanently, you should call
+    /// [`SortBuffer::delete`] manually.
+    pub fn consume(&mut self) -> impl Iterator<Item = DataChunk> + '_ {
```

Comment:
should we delete the records in state table here 🤔

Reply:
I guess it's hard for caller to use